### ISSUE TITLE
[BUG] The Admin Info view crash on client side in case of "tar" deployment : git metadata are missing

### ISSUE THREAD

**Issue by jdauphant** [RocketChat/Rocket.Chat#21794]
### Description:

The Admin Info view crash on client side

### Steps to reproduce:

1. Going to admin info page /admin/info

### Expected behavior:

Seeing server infos

### Actual behavior:

Quick flash of view infos and the view is fastly empty.
The result :
![image](https://user-images.githubusercontent.com/1238254/115992987-d1688280-a5d0-11eb-87c7-ae17b8c917d9.png)

### Server Setup Information:

- Version of Rocket.Chat Server: 3.13.3
- Operating System: Scalingo
- Deployment Method: scalingo
- Number of Running Instances: 1
- DB Replicaset Oplog: Enabled
- NodeJS Version: 12.20.2
- MongoDB Version: 4.0.3

### Client Setup Information

- Desktop App or Browser Version: Brave Version 1.17.75 Chromium: 87.0.4280.88 (Official Build) (64-bit) 
- Same error on : Firefox 87.0 (64-bit)
- Operating System: Pop OS

### Additional context

The impacted line is the code is : 
https://github.com/RocketChat/Rocket.Chat/blob/d6b97e849d4561d06ddeb2eb2ef3cf62170e814a/client/views/admin/info/DeploymentCard.js#L69 (Cannot read property 'slice' of undefined)

### Relevant logs:

Browser javascript console log : 
```
ui-kit version: 0.23.0
ad19555775301fc2b49963c6b08b4ed8eb00d86e.js?meteor_js_resource=true:1411 TypeError: Cannot read property 'slice' of undefined
    at eval (/client/views/admin/info/DeploymentCard.js:1)
    at S (/client/views/admin/info/DeploymentCard.js:1)
    at e (/client/views/admin/info/DeploymentCard.js:1)
    at ma (ad19555775301fc2b49963c6b08b4ed8eb00d86e.js?meteor_js_resource=true:1411)
    at rf (ad19555775301fc2b49963c6b08b4ed8eb00d86e.js?meteor_js_resource=true:1411)
    at ef (ad19555775301fc2b49963c6b08b4ed8eb00d86e.js?meteor_js_resource=true:1411)
    at ji (ad19555775301fc2b49963c6b08b4ed8eb00d86e.js?meteor_js_resource=true:1411)
    at Hi (ad19555775301fc2b49963c6b08b4ed8eb00d86e.js?meteor_js_resource=true:1411)
    at Ui (ad19555775301fc2b49963c6b08b4ed8eb00d86e.js?meteor_js_resource=true:1411)
    at Ri (ad19555775301fc2b49963c6b08b4ed8eb00d86e.js?meteor_js_resource=true:1411)
```

Server logs 
```
2021-04-25 13:35:33.760355906 +0200 CEST [web-1] Change Stream is available for your installation, give admin permissions to your database user to use this improved version.
2021-04-25 13:35:38.393376908 +0200 CEST [web-1] LocalStore: store created at
2021-04-25 13:35:38.436203884 +0200 CEST [web-1] LocalStore: store created at
2021-04-25 13:35:38.436472692 +0200 CEST [web-1] LocalStore: store created at
2021-04-25 13:35:38.801090807 +0200 CEST [web-1] Setting default file store to GridFS
2021-04-25 13:35:42.905323202 +0200 CEST [web-1] {"line":"120","file":"migrations.js","message":"Migrations: Not migrating, already at version 220","time":{"$date":1619350542904},"level":"info"}
2021-04-25 13:35:43.101850497 +0200 CEST [web-1] ufs: temp directory created at "/tmp/ufs"
2021-04-25 13:35:44.775443357 +0200 CEST [web-1] Loaded the Apps Framework and loaded a total of 0 Apps!
2021-04-25 13:35:44.850933910 +0200 CEST [web-1] Using GridFS for custom sounds storage
2021-04-25 13:35:44.852647686 +0200 CEST [web-1] Using GridFS for custom emoji storage
2021-04-25 13:35:45.049876381 +0200 CEST [web-1] Browserslist: caniuse-lite is outdated. Please run next command `npm update`
2021-04-25 13:35:45.242359386 +0200 CEST [web-1] Updating process.env.MAIL_URL
2021-04-25 13:35:45.495424519 +0200 CEST [web-1] ➔ System ➔ startup
2021-04-25 13:35:45.495766736 +0200 CEST [web-1] ➔ +----------------------------------------------------------+
2021-04-25 13:35:45.495996004 +0200 CEST [web-1] ➔ |                      SERVER RUNNING                      |
2021-04-25 13:35:45.496176156 +0200 CEST [web-1] ➔ +----------------------------------------------------------+
2021-04-25 13:35:45.496774538 +0200 CEST [web-1] ➔ |  Rocket.Chat Version: 3.13.3                             |
2021-04-25 13:35:45.496491077 +0200 CEST [web-1] ➔ |                                                          |
2021-04-25 13:35:45.497156027 +0200 CEST [web-1] ➔ |      MongoDB Version: 4.0.3                              |
2021-04-25 13:35:45.496970401 +0200 CEST [web-1] ➔ |       NodeJS Version: 12.20.2 - x64                      |
2021-04-25 13:35:45.497904757 +0200 CEST [web-1] ➔ |             Site URL: https://rocketchat.incubateur.net  |
2021-04-25 13:35:45.497781085 +0200 CEST [web-1] ➔ |         Process Port: 28537                              |
2021-04-25 13:35:45.497577177 +0200 CEST [web-1] ➔ |             Platform: linux                              |
2021-04-25 13:35:45.497380098 +0200 CEST [web-1] ➔ |       MongoDB Engine: unknown                            |
2021-04-25 13:35:45.498604338 +0200 CEST [web-1] ➔ +----------------------------------------------------------+
2021-04-25 13:35:45.498432326 +0200 CEST [web-1] ➔ |                                                          |
2021-04-25 13:35:45.498091481 +0200 CEST [web-1] ➔ |     ReplicaSet OpLog: Enabled                            |
```
commit.hash isn't shown : https://github.com/RocketChat/Rocket.Chat/blob/157b882d38b34b7b79dea527316554fe33f70891/server/startup/serverRunning.js#L41

**Comment by apurv-1** [comment]
Hey @jdauphant, I was not able to reproduce the bug, probably because I'm running Rocket.Chat Version: 3.14.0-develop. Please update the branch.

**Comment by jdauphant** [comment]
Ok, i have try the 3.14.0-rc1 (this commit : https://github.com/RocketChat/Rocket.Chat/commit/ed109dae4d0b97f67222a8f0720f7d3c75d2a725 ). The problem is worst : (the admin menu also disapear)
![image](https://user-images.githubusercontent.com/1238254/116155950-226c9980-a6eb-11eb-9b9b-dc4b6c0de0dc.png)
Browser Side :
```
ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398 TypeError: Cannot read property 'slice' of undefined
    at eval (/client/views/admin/info/DeploymentCard.js:1)
    at S (/client/views/admin/info/DeploymentCard.js:1)
    at e (/client/views/admin/info/DeploymentCard.js:1)
    at ja (ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398)
    at di (ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398)
    at si (ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398)
    at Mf (ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398)
    at rs (ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398)
    at $f (ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398)
    at Zf (ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398)
Di @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
Mi.t.callback @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
Tc @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
Hi @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
os @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
t.unstable_runWithPriority @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1416
gc @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
ns @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
Lf @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
(anonymous) @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
t.unstable_runWithPriority @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1416
gc @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
_c @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
bc @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
Of @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
$a @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
t @ /client/views/admin/info/InformationRoute.js:1
async function (async)
t @ /client/views/admin/info/InformationRoute.js:1
eval @ /client/views/admin/info/InformationRoute.js:1
ss @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
t.unstable_runWithPriority @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1416
gc @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
as @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
Lf @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
(anonymous) @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
t.unstable_runWithPriority @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1416
gc @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
_c @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
bc @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
Of @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
$a @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398
F.i.current @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1365
setTimeout (async)
(anonymous) @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1365
(anonymous) @ ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1365
/client/views/admin/info/InformationRoute.js:1 Uncaught (in promise) TypeError: Cannot read property 'slice' of undefined
    at eval (/client/views/admin/info/DeploymentCard.js:1)
    at S (/client/views/admin/info/DeploymentCard.js:1)
    at e (/client/views/admin/info/DeploymentCard.js:1)
    at ja (ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398)
    at di (ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398)
    at si (ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398)
    at Mf (ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398)
    at rs (ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398)
    at $f (ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398)
    at Zf (ccf99aa3f37ef600f1c66e306576d6741dd8dee4.js?meteor_js_resource=true:1398)
```
Server Side : 
```
2021-04-26 23:54:58.976648534 +0200 CEST[web-1] Change Stream is available for your installation, give admin permissions to your database user to use this improved version.
2021-04-26 23:55:04.035830390 +0200 CEST[web-1] LocalStore: store created at
2021-04-26 23:55:04.085864159 +0200 CEST[web-1] LocalStore: store created at
2021-04-26 23:55:04.086051816 +0200 CEST[web-1] LocalStore: store created at
2021-04-26 23:55:04.458371755 +0200 CEST[web-1] Setting default file store to GridFS
2021-04-26 23:55:08.748865829 +0200 CEST[web-1] {"line":"120","file":"migrations.js","message":"Migrations: Migrating from version 220 -> 222","time":{"$date":1619474108747},"level":"info"}
2021-04-26 23:55:08.750246180 +0200 CEST[web-1] {"line":"120","file":"migrations.js","message":"Migrations: Running up() on version 221","time":{"$date":1619474108749},"level":"info"}
2021-04-26 23:55:08.757167932 +0200 CEST[web-1] {"line":"120","file":"migrations.js","message":"Migrations: Running up() on version 222","time":{"$date":1619474108756},"level":"info"}
2021-04-26 23:55:08.762292358 +0200 CEST[web-1] {"line":"120","file":"migrations.js","message":"Migrations: Finished migrating.","time":{"$date":1619474108761},"level":"info"}
2021-04-26 23:55:18.763474072 +0200 CEST[web-1] Not migrating, control is locked. Attempt 1/30. Trying again in 10 seconds.
2021-04-26 23:55:18.769010436 +0200 CEST[web-1] {"line":"120","file":"migrations.js","message":"Migrations: Not migrating, already at version 222","time":{"$date":1619474118768},"level":"info"}
2021-04-26 23:55:18.992054327 +0200 CEST[web-1] ufs: temp directory created at "/tmp/ufs"
2021-04-26 23:55:20.932697572 +0200 CEST[web-1] Loaded the Apps Framework and loaded a total of 0 Apps!
2021-04-26 23:55:20.938480737 +0200 CEST[web-1] Setting up email interceptor for test-inbox-rocket-chat@beta.gouv.fr
2021-04-26 23:55:21.037039999 +0200 CEST[web-1] Using GridFS for custom sounds storage
2021-04-26 23:55:21.040045038 +0200 CEST[web-1] Using GridFS for custom emoji storage
2021-04-26 23:55:21.395678477 +0200 CEST[web-1] Updating process.env.MAIL_URL
2021-04-26 23:55:21.898031007 +0200 CEST[web-1] ➔ System ➔ startup
2021-04-26 23:55:21.898429838 +0200 CEST[web-1] ➔ +----------------------------------------------------------+
2021-04-26 23:55:21.898751104 +0200 CEST[web-1] ➔ | SERVER RUNNING |
2021-04-26 23:55:21.898911345 +0200 CEST[web-1] ➔ +----------------------------------------------------------+
2021-04-26 23:55:21.899088385 +0200 CEST[web-1] ➔ | |
2021-04-26 23:55:21.899223964 +0200 CEST[web-1] ➔ | Rocket.Chat Version: 3.14.0-develop |
2021-04-26 23:55:21.899430341 +0200 CEST[web-1] ➔ | NodeJS Version: 12.20.2 - x64 |
2021-04-26 23:55:21.899599822 +0200 CEST[web-1] ➔ | MongoDB Version: 4.0.3 |
2021-04-26 23:55:21.899777529 +0200 CEST[web-1] ➔ | MongoDB Engine: unknown |
2021-04-26 23:55:21.899929656 +0200 CEST[web-1] ➔ | Platform: linux |
2021-04-26 23:55:21.900247298 +0200 CEST[web-1] ➔ | Site URL: https://rocketchat.incubateur.net |
2021-04-26 23:55:21.900098039 +0200 CEST[web-1] ➔ | Process Port: 29662 |
2021-04-26 23:55:21.900564659 +0200 CEST[web-1] ➔ | |
2021-04-26 23:55:21.900434509 +0200 CEST[web-1] ➔ | ReplicaSet OpLog: Enabled |
2021-04-26 23:55:21.900704943 +0200 CEST[web-1] ➔ +----------------------------------------------------------+
2021-04-26 23:55:52.344570478 +0200 CEST[web-1] (node:15) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
```

Also I deploy on Scalingo with your scalingo integration (without modification), here is the full log of the building process, if it can be usefull : http://pastebin.fr/90526

**Comment by jdauphant** [comment]
@apurv-1 I have one assuption, deploy are made by "tar" apparently (so maybe git metadata are not available in that case)
![image](https://user-images.githubusercontent.com/1238254/116158103-5eedc480-a6ee-11eb-949a-3cac9bd4522e.png)

But the view DeploymentCard.js assume their is always git metadata available. https://github.com/RocketChat/Rocket.Chat/blob/d6b97e849d4561d06ddeb2eb2ef3cf62170e814a/client/views/admin/info/DeploymentCard.js#L69
We can see here : https://github.com/RocketChat/Rocket.Chat/blob/157b882d38b34b7b79dea527316554fe33f70891/packages/rocketchat-version/plugin/compile-version.js#L33
That the server is ok for not having a hash, date, author and subject (it's the source of the data use in the server, that the client retrieve by api).

You may reproduce it if you try to build and launch without ".git", if it's the case, the "admin/info" page may fail in all deployement without the git metadata.

**Comment by alemangui** [comment]
Any news on this issue?

**Comment by gabriellsh** [comment]
This has been fixed a while ago by #22057. I believe that from version 3.15 onwards this should no longer be an issue. Closing this.
